src/components/NewsFeedView.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { RefreshableFeed } from './RefreshableFeed';
+import { Feed } from '../models/Feed';
+import { Post } from '../models/Post';
+import { Settings } from '../models/Settings';
+import { NavigationHeader } from './NavigationHeader';
+
+export interface DispatchProps {
+    onRefreshPosts: (feeds: Feed[]) => void;
+}
+
+export interface StateProps {
+    navigation: any;
+    posts: Post[];
+    feeds: Feed[];
+    settings: Settings;
+}
+
+type Props = StateProps & DispatchProps;
+
+export const NewsFeedView = (props: Props) => {
+    return (
+        <RefreshableFeed {...props}>
+            {{
+                listHeader: <NavigationHeader
+                                leftButtonText=''
+                                title='News'
+                            />,
+            }}
+        </RefreshableFeed>
+    );
+};
+
+export const MemoizedNewsFeedView = React.memo(NewsFeedView);
```

Comment:
As far as I understand, right now we don't gain anything with memoization, because we use `immutable-js` and we generate the props from it any time the state changes. 

One solution could be to get rid of `immutable-js`. Recently I tend to like this idea, because there is not a lot we gain by using it, but we have several problems with it (like this one, or the incorrect type defs).

Reply:
Also because the serialization of immutable Lists and Maps are stupidly inefficient (it doubles the size of the backup compared to plain JSON).

Reply:
I would keep the memoized version, because after getting rid of immutable it will be better. Also I would change the names so that NewsFeedView is the memoized version and there would be a NotMemoizedNewsFeed (or some better name but you get the point :)

Reply:
the classic discussion about IServiceProvider + ServiceProvider OR ServiceProvider + DefaultServiceProvider/ServiceProviderImpl...
I have no opinion, so be it :-)